portal-api/src/v1/users/sign-in-link.service.js

```diff
@@ -0,0 +1,69 @@
+const { ObjectId } = require('mongodb');
+const crypto = require('node:crypto');
+const sendEmail = require('../email');
+const { EMAIL_TEMPLATE_IDS, SIGN_IN_LINK_EXPIRY_MINUTES } = require('../../constants');
+const db = require('../../drivers/db-client');
+
+module.exports.createAndEmailSignInLink = async (user) => {
+  const {
+    _id: userId,
+    email: userEmail,
+    firstname: userFirstName,
+    surname: userLastName,
+  } = user;
+
+  const signInCode = await createSignInCode();
+  await saveSignInCodeHashAndSalt({ userId, signInCode });
+  return sendSignInLinkEmail({
+    userEmail,
+    userFirstName,
+    userLastName,
+    // TODO DTFS2-6750: introduce an environment variable to replace localhost
+    signInLink: `http://localhost/login/sign-in-link?t=${signInCode}`,
+  });
+};
+
+async function createSignInCode() {
+  try {
+    return crypto.randomBytes(32).toString('hex');
+  } catch (e) {
+    const error = new Error('Failed to create a sign in code.');
+    error.cause = e;
+    throw error;
+  }
+}
+
+async function saveSignInCodeHashAndSalt({ userId, signInCode }) {
+  try {
+    const salt = crypto.randomBytes(64);
```

Comment:
I'm not sure what you feel about this -- Do you think we should extract salt and hash generation to `utils.js`?

Reply:
I agree that it feels too re-usable / to separate a concern to be _here_
I dislike the `crypto/utils.js` file because it's too much of a catch all. I'll see what I can do